smartsim/entity/dbobject.py

```diff
@@ -107,18 +109,24 @@ def _enumerate_devices(self) -> t.List[str]:
         :rtype: list[str]
         """
         devices = []
-        if ":" in self.device and self.devices_per_node > 1:
-            msg = "Cannot set devices_per_node>1 if a device numeral is specified, "
-            msg += f"the device was set to {self.device} and devices_per_node=={self.devices_per_node}"
-            raise ValueError(msg)
-        if self.device in ["CPU", "GPU"] and self.devices_per_node > 1:
+        if self.device in ["GPU"] and self.devices_per_node > 1:
             for device_num in range(self.devices_per_node):
```

Comment:
I'm a big fan of declarative style. Consider a list comprehension instead of loops:

```
devices = [f"{self.device}:{index}" for index in range(self.devices_per_node)]
```

This also makes it clear we're _not_ doing multiple appends from multiple branches of the `if`


You could also refactor to avoid nesting and significantly shorten:

```
if self.device == "GPU" and self.devices_per_node > 1:
  return [f"{self.device}:{index}" for index in range(self.devices_per_node)]

return [self.device]
```